### DISCUSSION TITLE
How to correct this script

### DISCUSSION THREAD

**Discussion by DanyanXu** [windingwind/zotero-actions-tags#122] [Q&A] [answered]
There is a script about merge the duplicate item. I wonder how to use it by this addon. I have tried like below, but it didn't work.
![image](https://github.com/windingwind/zotero-actions-tags/assets/84708952/e907dedf-c650-4566-830d-2fafd3db953f)
```
var DupPane = Zotero.getZoteroPanes();
for(var i = 0; i < 30000; i++) {await new Promise(r => setTimeout(r, 1000));
// 选择较旧的版本作为主条目
Zotero_Duplicates_Pane.setMaster(0);
DupPane[0].mergeSelectedItems();
Zotero_Duplicates_Pane.merge();}
```

**Comment by windingwind** [comment, marked as answer]
```js
const ZoteroPane = require("ZoteroPane");
const Zotero_Duplicates_Pane = require("Zotero_Duplicates_Pane");
// 选择较旧的版本作为主条目
Zotero_Duplicates_Pane.setMaster(0);
ZoteroPane.mergeSelectedItems();
Zotero_Duplicates_Pane.merge();
```

Are you trying to repeatedly execute something? That's dangerous.

**Reply by DanyanXu** [reply]
YES. I don't want to manually merge duplicate items. Can you help me correct the code above? Thanks!

**Reply by windingwind** [reply]
You can merge all at once, but I don't think repeatedly running in the background is a good idea.
The above code should be able to merge the selected item.

**Reply by DanyanXu** [reply]
Would you have a plan to write a script about merging the duplicate items? This code doesn't work. TT  Thank you a lot for your answer and your addon!

**Reply by windingwind** [reply]
sorry, I can't help

**Reply by DanyanXu** [reply]
It doesn't matter. Thank you a lot.
